### PR TITLE
Trap exception patch

### PR DESCRIPTION
**Pull request by h4570**

### CODE REVIEW

File: .vscode/configurations/windows-docker/tasks.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
         {
             "type": "shell",
             "label": "1. Copy source code to docker volume",
-            "command": "docker exec -t tyra-game-compiler sh -c 'rsync -ac --delete --exclude \".git\" --exclude \".vscode\" --exclude \"*obj*/**\" --exclude \"*bin*/**\" /host/ /src/'"
+            "command": "docker exec -t tyra-game-compiler sh -c 'rsync -ac --delete --exclude \".git\" --exclude \".vscode\" --exclude \"obj/**\" --exclude \"bin/**\" /host/ /src/'"
```
Comment (h4570) [inline .vscode/configurations/windows-docker/tasks.json:21]: There was 2 problems with docker:
1 - `res/` folder was not copied to output directory. This was a Tyra bug, so please run "Update Tyra" task

2 - There was problem with copying binaries (problem `file has vanished..`), so please apply these configuration patches

File: inc/managers/state_manager.hpp [hunk 2]
```diff
@@ -25,10 +25,9 @@ class StateManager {
   StateManager();
   ~StateManager();
 
-  Sprite* sprite = new Sprite;
```
Comment (h4570) [inline inc/managers/state_manager.hpp:28]: It is better to allocate/deallocate in constructor

File: src/tyracraft_game.cpp [hunk 3]
```diff
@@ -19,7 +19,7 @@ void TyraCraftGame::init() {
 void TyraCraftGame::loop() {
   engine->renderer.beginFrame(CameraInfo3D(&camera.position, &camera.lookPos));
   printf("%i\n", engine->info.getFps());
-  stateManager.update(engine, 1 / engine->info.getFps(), &engine->pad, &camera);
+  stateManager.update(1 / static_cast<float>(engine->info.getFps()), &camera);
```
Comment (h4570) [inline src/tyracraft_game.cpp:22]: This was the core problem.
You was expecting `float` as parameter, but `u32` was given.
`1 / engine->info.getFps()` -> `int / int` -> `int`.

In c++ if one of the numbers is `float`, the result will be `float`
so:
`1 / static_cast<float>(engine->info.getFps())` -> `int / float` -> `float`

File: .vscode/configurations/windows-docker/tasks.json (hunk 1 above)
Comment (Wellinator) [reply .vscode/configurations/windows-docker/tasks.json:21]: I saw that bugs, but I thougth I did changed something I soudn't... Thx for fixing!

File: inc/managers/state_manager.hpp (hunk 2 above)
Comment (Wellinator) [reply inc/managers/state_manager.hpp:28]: Yeah! THe entyre code will be refactored xD

File: src/tyracraft_game.cpp (hunk 3 above)
Comment (Wellinator) [reply src/tyracraft_game.cpp:22]: Okay, I'll take a deeper look into that behaviour! Thx for that!

**Pull request merged**